src/main/scala/cromwell/engine/workflow/SingleWorkflowRunnerActor.scala

```diff
@@ -79,18 +85,19 @@ case class SingleWorkflowRunnerActor(source: WorkflowSourceFiles,
       log.info(s"$tag: workflow ID UUID($id)")
       workflowManager ! SubscribeToWorkflow(id)
 
-      if (getAbortJobsOnTerminate) {
-        Runtime.getRuntime.addShutdownHook(new Thread() {
-          override def run(): Unit = {
-            workflowManager ! WorkflowAbort(id)
-            log.info(s"$tag: Waiting for workflow $id to abort...")
-            while(stateName != Done)
-              Thread.sleep(1000)
-            log.info(s"$tag: Workflow $id aborted.")
-          }
-        })
+      if (abortJobsOnTerminate) {
+        sys.addShutdownHook {
+          workflowManager ! WorkflowAbort(id)
+          log.info(s"$tag: Waiting for workflow $id to abort...")
+          Await.ready(donePromise.future, Duration.Inf)
```

Comment:
This donePromise seems incredibly funky to me. 

It's basically used like a boolean var, but hidden in a Promise to avoid offending FP sensibilities?